cpu/samd21/periph/rtt.c

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (C) 2015 Daniel Krebs
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_samd21
+ * @{
+ * @file
+ * @brief       Low-level RTT driver implementation
+ * @author      Daniel Krebs <github@daniel-krebs.net>
+ * @}
+ */
+
+#include <time.h>
+#include "cpu.h"
+#include "periph/rtt.h"
+#include "periph_conf.h"
+#include "sched.h"
+#include "thread.h"
+
+/* guard file in case no RTT device was specified */
+#if RTT_NUMOF
+
+
+typedef struct {
+    rtt_cb_t    overflow_cb;    /**< called from RTT interrupt on overflow */
+    void*       overflow_arg;   /**< argument passed to overflow callback */
+
+    rtt_cb_t    alarm_cb;       /**< called from RTT interrupt on alarm */
+    void*       alarm_arg;      /**< argument passen to alarm callback */
+} rtt_state_t;
+
+static rtt_state_t rtt_callback;
+
+
+/**
+ * @brief Initialize RTT module
+ *
+ * The RTT is running at 32768 Hz by default, i.e. @ XOSC32K frequency without
+ * divider. There are 2 cascaded dividers in the clock path:
+ *
+ *  - GCLK_GENDIV_DIV(n): between 1 and 31
+ *  - RTC_MODE0_CTRL_PRESCALER_DIVn: between 1 and 1024, see defines in `component_rtc.h`
+ */
```

Comment:
Should add:
``- GCLK_GENCTRL_DIVSEL = 0, clock get divided by GENDIV.DIV; 1, clock get divided by  2^(GENDIV.DIV+1)``

With this, it allows to get 1Hz